x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/io/stream/PlanNamedTypes.java

```diff
@@ -793,24 +792,14 @@ static void writeDissect(PlanStreamOutput out, Dissect dissect) throws IOExcepti
     }
 
     static EsRelation readEsRelation(PlanStreamInput in) throws IOException {
-        Source source = in.readSource();
-        EsIndex esIndex = readEsIndex(in);
-        List<Attribute> attributes = readAttributes(in);
-        EsSourceOptions esSourceOptions = in.getTransportVersion().onOrAfter(TransportVersions.ESQL_ES_SOURCE_OPTIONS)
```

Comment:
I think we need to keep serialization around (read/write but discard) to maintain bwc for users who do not use OPTIONS.

Reply:
Thanks, indeed, we need to satisfy the serialisation requirements between `TransportVersions.ESQL_ES_SOURCE_OPTIONS` and "now" -- newly introduced `TransportVersions.ESQL_REMOVE_ES_SOURCE_OPTIONS`.